ownCloudAppShared/AppLock/AppLockManager.swift

```diff
@@ -381,6 +390,21 @@ public class AppLockManager: NSObject {
 	}
 
 	// MARK: - Status
+	
+	private var isLockBypassed: Bool {
+	 if let lockedUntilDate = lockedUntilDate, let lockSetTime = lockSetTime {
+		 let currentTime = ProcessInfo.processInfo.systemUptime
+		 let lockDuration = currentTime - lockSetTime
+		 let timeRemaining = lockedUntilDate.timeIntervalSinceNow
+
+		 if lockDuration < timeRemaining {
```

Comment:
- `lockDuration` computes the time _since_ the lock has been set
- `timeRemaining` computes the time _until_ the lock expires

Comparing the two will not be able to tell if there was an attempt to bypass the lock. Image a 10 second delay, where 2 seconds have passed by and 8 seconds are remaining - without any bypass attempt. These numbers would fit the condition for a lock-bypass above even though it hasn't been bypassed.

To determine a bypass, it's necessary to also keep track of the date/time the lock has been put in place - and then use that date to compute the time passed since locking, to compare it against the time passed in system uptime.
That comparison should also leave a tiny bit of wiggle-room (f.ex. ~ 0.05 seconds), because the systemUptime and the date are not taken at the exact same moment and therefore will regularly differ slightly.